Accept arrays in parse_config's method_missing (#5252)
chef/inspec: docs-chef-io/content/inspec/resources/parse_config.md

- May run a command to `stdout`, and then run the test against that output
- May use options to define how configuration data is to be parsed

## Options

> This resource supports the following options for parsing configuration data. Use them in an `options` block stated outside of (and immediately before) the actual test:

Reviewer: ```suggestion
This resource supports multiple options to parse configuration data. Use the options in an `options` block stated outside of (and immediately before) the actual test. For example:
```

    options = {
        assignment_regex: /^\s*([^:]*?)\s*:\s*(.*?)\s*$/,
        multiple_values: true
      }

>     output = command('some-command').stdout

Reviewer 2: We should use this new example code; but bear in mind that we increasingly see issues with using Ruby variable assignment like this, because the code executes unconditionally even if the control is skipped/waived/excluded.

The only current solution is to write a custom resource that wraps up the code. In future we want InSpec to to properly skip code for excluded controls...

    'key = value # comment'
    params['key'] = 'value'

## Examples

> This resource is based on the `parse_config_file` resource. As such please refer to the examples of the [`parse_config_file`](/inspec/resources/parse_config_file) resource.

Reviewer: ```suggestion
This resource is based on the `parse_config_file` resource. See the [`parse_config_file`](/inspec/resources/parse_config_file) resource for examples.
```
